twitter/twitter_client.bal

```diff
@@ -0,0 +1,1057 @@
+    # + return - Returns Followers object when successful, else returns an error.
+    public remote function getFollowers(int? userId = (), string? screenName = (), int? cursor = (), int? count = (),
+                                        boolean? skipStatus = (), boolean? includeUserEntities = ())
+                               returns Followers|error;
+};
+
+# Twitter Trends Client object.
+#
+# + httpClient - HTTP Client.
+# + twitterClient - Twitter Client.
+public type TwitterTrendsClient client object {
+    http:Client httpClient;
+    TwitterClient|error twitterClient;
+
+    public function __init(TwitterClient|TwitterConfiguration config) returns error?{
+        if (config is TwitterClient) {
+            self.httpClient = config.getHttpClient();
+            self.twitterClient = config;
+        } else {
+            self.httpClient = new(TWITTER_API_URL, config = config.clientConfig);
+            var isValid = verifyCredentials(self.httpClient, config.clientId, config.clientSecret, config.accessToken,
+                config.accessTokenSecret);
+            if (isValid is error) {
+                return isValid;
+            }
+            self.twitterClient = new(config);
+        }
+    }
+
+    # Get the Twitter client to invoke tweet related functions.
+    #
+    # + return - Returns `TwitterClient` object.
+    public function getTwitterClient() returns TwitterClient? {
+        TwitterClient|error twitterClient = self.twitterClient;
+        if (twitterClient is TwitterClient) {
+            return twitterClient;
+        }
+        return ();
+    }
+
+    # Returns the top trending topics for a specific WOEID, if trending information is available for it.
+    #
+    # + locationId - The ID of the location to return trending information for.
+    # + exclude - Setting this equal to hashtags will remove all hashtags from the trends list.
+    # + return - Returns a collection of TrendsList object when successful, else returns an error.
+    public remote function getTrendsByPlace(int locationId, string? exclude = ()) returns TrendsList[]|error;
+
+    # Returns the locations that Twitter has trending topic information for, closest to a specified location.
+    #
+    # + latitude - Latitude of the location.
+    # + longitude - Longitude of the location.
+    # + return - Returns a collection of Location object when successful, else returns an error.
+    public remote function getClosestTrendLocations(float latitude, float longitude) returns Location[]|error;
+};
+
+public remote function TwitterClient.tweet(string status, string? cardUri = (), boolean? trimUser = (),
+                                           boolean? enableDmcommands = (), boolean? failDmcommands = (),
+                                           MediaParams? mediaParams = (), LocationParams? locationParams = ())
+                                         returns Tweet|error {
+    http:Client httpClient = self.httpClient;
```

Comment:
Why are we doing this assignment to a new variable without passing `self.httpClient`?